src/describe/sobjectApi.ts

```diff
@@ -0,0 +1,319 @@
+  calculated: boolean;
+  calculatedFormula?: any;
+  cascadeDelete: boolean;
+  caseSensitive: boolean;
+  compoundFieldName?: any;
+  controllerName?: any;
+  createable: boolean;
+  custom: boolean;
+  defaultValue?: boolean;
+  defaultValueFormula?: any;
+  defaultedOnCreate: boolean;
+  dependentPicklist: boolean;
+  deprecatedAndHidden: boolean;
+  digits: number;
+  displayLocationInDecimal: boolean;
+  encrypted: boolean;
+  externalId: boolean;
+  extraTypeInfo?: any;
+  filterable: boolean;
+  filteredLookupInfo?: any;
+  groupable: boolean;
+  highScaleNumber: boolean;
+  htmlFormatted: boolean;
+  idLookup: boolean;
+  inlineHelpText?: any;
+  label: string;
+  length: number;
+  mask?: any;
+  maskType?: any;
+  name: string;
+  nameField: boolean;
+  namePointing: boolean;
+  nillable: boolean;
+  permissionable: boolean;
+  picklistValues: any[];
+  polymorphicForeignKey: boolean;
+  precision: number;
+  queryByDistance: boolean;
+  referenceTargetField?: any;
+  referenceTo: string[];
+  relationshipName: string;
+  relationshipOrder?: any;
+  restrictedDelete: boolean;
+  restrictedPicklist: boolean;
+  scale: number;
+  searchPrefilterable: boolean;
+  soapType: string;
+  sortable: boolean;
+  type: string;
+  unique: boolean;
+  updateable: boolean;
+  writeRequiresMasterRead: boolean;
+}
+/* tslint:enable:no-any */
+
+export interface Urls {
+  layout: string;
+}
+
+export interface RecordTypeInfo {
+  active: boolean;
+  available: boolean;
+  defaultRecordTypeMapping: boolean;
+  master: boolean;
+  name: string;
+  recordTypeId: string;
+  urls: Urls;
+}
+
+export interface SupportedScope {
+  label: string;
+  name: string;
+}
+
+export interface Urls2 {
+  compactLayouts: string;
+  rowTemplate: string;
+  approvalLayouts: string;
+  uiDetailTemplate: string;
+  uiEditTemplate: string;
+  defaultValues: string;
+  describe: string;
+  uiNewRecord: string;
+  quickActions: string;
+  layouts: string;
+  sobject: string;
+}
+
+export interface DescribeSObjectResult {
+  sObjectName: string;
+  result?: SObject;
+  timestamp?: string;
+}
+
+export enum SObjectCategory {
+  ALL = 'ALL',
+  STANDARD = 'STANDARD',
+  CUSTOM = 'CUSTOM'
+}
+
+export type SubRequest = { method: string; url: string };
+export type BatchRequest = { batchRequests: SubRequest[] };
+export type DescribeResponse = {
+  statusCode: number;
+  headers: { [key: string]: string };
+  body: string;
+};
+export type SubResponse = { statusCode: number; result: SObject };
+export type BatchRawResponse = {
+  statusCode: number;
+  headers: { [key: string]: string };
+  body: string;
+};
+export type BatchResponse = { hasErrors: boolean; results: SubResponse[] };
+
+export class SObjectDescribeAPI {
+  private readonly connection: Connection;
+
+  // URL constants
+  private readonly SERVICESPATH: string = 'services/data';
+  // the targetVersion should be consistent with the Cli even if only using REST calls
+  private readonly TARGETVERSION = '49.0';
```

Comment:
Why are you hardcoding the version? This should just use `connection.getApiVersion`. https://github.com/forcedotcom/sfdx-core/blob/develop/src/connection.ts#L199

Reply:
Also if you are part of the connection class, there are some methods that build part of this request for you, so you don't have to piece it all together, like services/data, etc.